### PR TITLE
536 cr creation sort the wbs dropdown

### PR DESCRIPTION
**Pull request by superhvarn**
## Changes

Added functionality for the CR create page to sort its dropdown for projects by wbs number

## Screenshots
Screen Shot 2023-09-11 at 9 46 59 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.

Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #536)

### CODE REVIEW

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/shared/src/validate-wbs.ts [hunk 1]
```diff
@@ -5,6 +5,28 @@
 
 import { WbsNumber } from './types/project-types';
 
+export const wbsNumComparator = (wbsNum1: any, wbsNum2: any) => {
```
Comment (z-roth) [inline src/shared/src/validate-wbs.ts:8]: let's avoid using the `any` type here

**Comment by z-roth** [comment]
@RChandler234 thoughts on including tests for the comparator?

File: src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequestView.tsx [hunk 2]
```diff
@@ -122,6 +124,8 @@ const CreateChangeRequestsView: React.FC<CreateChangeRequestViewProps> = ({
     });
   });
 
+  wbsDropdownOptions.sort(wbsNumComparator);
```
Comment (RChandler234) [inline src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequestView.tsx:127]: You need to pass in a method into sort like this: .sort((a,b) => comparator(a.id, b.id) so you don't have to take out the ids in the comparator

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Comment by RChandler234** [comment]
> @RChandler234 thoughts on including tests for the comparator?

Ooo like unit testing our util methods, that's a good idea (would make a lot of good tickets as well)

**Comment by z-roth** [comment]
@superhvarn lets add a test here for the comparator as well

**Comment by z-roth** [comment]
@superhvarn lets add a test for the comparator here as well

File: src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequestView.tsx [hunk 3]
```diff
@@ -122,6 +124,8 @@ const CreateChangeRequestsView: React.FC<CreateChangeRequestViewProps> = ({
     });
   });
 
+  wbsDropdownOptions.sort((a, b) => wbsNumComparator(a.id, b.id));
```
Comment (RChandler234) [inline src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequestView.tsx:127]: we should also rename these from a and b to wbsNum1, wbsNum2 to be more precise/accurate

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/frontend/src/tests/utils/validate-wbs.test.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { wbsNumComparator } from 'shared';
+import '@testing-library/jest-dom/extend-expect';
+
+describe('wbsNumComparator', () => {
+  it('should correctly compare two numbers', () => {
+    expect(wbsNumComparator('1.1.3', '1.3.4')).toBe(-1);
+    expect(wbsNumComparator('1.2.1', '1.2.1')).toBe(0);
+    expect(wbsNumComparator('1.3.4', '1.1.3')).toBe(1);
```
Comment (z-roth) [inline src/frontend/src/tests/utils/validate-wbs.test.ts:8]: can we add a case for differing car numbers? (the first digit)

**Review by z-roth — CHANGES_REQUESTED** [review]

**Review by z-roth — APPROVED** [review]

**Comment by z-roth** [comment]
looks good to me! 💸

**Review by walker-sean — APPROVED** [review]
LGTM 🔥

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/shared/src/validate-wbs.ts [hunk 5]
```diff
@@ -5,6 +5,27 @@
 
 import { WbsNumber } from './types/project-types';
 
+/**
+ * Compares two wbs numbers
```
Comment (walker-sean) [inline src/shared/src/validate-wbs.ts:9]: Add that it's used to sort WBS nums in ascending order to make it extra clear

**Review by walker-sean — DISMISSED** [review]

**Review by RChandler234 — APPROVED** [review]

**Pull request merged**